src/result.ts

```diff
@@ -308,3 +315,171 @@ export class Err<T, E> implements IResult<T, E> {
 }
 
 export const fromThrowable = Result.fromThrowable
+
+//#region Combine - Types
+
+// This is a helper type to prevent infinite recursion in typing rules.
+//
+// Use this with your `depth` variable in your types.
+type Prev = [
+  never,
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+  6,
+  7,
+  8,
+  9,
+  10,
+  11,
+  12,
+  13,
+  14,
+  15,
+  16,
+  17,
+  18,
+  19,
+  20,
+  ...0[]
+]
+
+// Collects the results array into separate tuple array.
+//
+// T         - The array of the results
+// Collected - The collected tuples.
+// Depth     - The maximum depth.
+type CollectResults<T, Collected extends unknown[] = [], Depth extends number = 5> = [
+  Depth,
+] extends [never]
+  ? []
+  : T extends [infer H, ...infer Rest]
+  ? // And test whether the head of the list is a result
+    H extends Result<infer L, infer R>
+    ? // Continue collecting...
+      CollectResults<
+        // the rest of the elements
+        Rest,
+        // The collected
+        [...Collected, [L, R]],
+        // and one less of the current depth
+        Prev[Depth]
+      >
+    : never // Impossible
+  : Collected
+
+// Transposes an array
+//
+// A          - The array source
+// Transposed - The collected transposed array
+// Depth      - The maximum depth.
+type Transpose<A, Transposed extends unknown[][] = [], Depth extends number = 10> = A extends [
```

Comment:
Could we export this to have 1 or 2 type-level tests written for it, please?

Reply:
@supermacro